traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go

```diff
@@ -188,14 +188,30 @@ func (cg *TOCacheGroup) ValidateTypeInTopology() error {
 		return nil
 	}
 
+	// language=SQL
+	const previousNameQuery = `
+	SELECT name
+	FROM cachegroup c
+	WHERE c.id = $1
+	`
+	var previousName string
+	err = cg.ReqInfo.Tx.QueryRow(previousNameQuery, *cg.ID).Scan(&previousName)
+	if err == sql.ErrNoRows {
+		return nil
+	}
+	if err != nil {
+		log.Errorf("%s: getting the previous name of cachegroup %s: %s", userErr.Error(), *cg.Name, err.Error())
```

Comment:
This line panics  if `userErr` is nil. Changing `userErr.Error()` to `userErr` would fix that

Reply:
Doesn't this line prevent that?
https://github.com/apache/trafficcontrol/blob/c0b747c18799952a4db7566c5a0f5dd9a5989c95/traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go#L164

Reply:
oh, true